branchpro/apps/_simulation.py

```diff
@@ -248,36 +345,49 @@ def update_simulation(self, new_init_cond, new_r0, new_r1, new_t1):
         new_t1
             (float) updated position on the slider for the time change in
             reproduction numbers for the Branch Pro model in the simulator.
+        new_epsilon
+            (float) updated position on the slider for the constant of
+            proportionality between local and imported cases for the Branch Pro
+            model in the posterior.
 
         Returns
         -------
         pandas.DataFrame
             Simulations storage dataframe
         """
         data = self.session_data.get('data_storage')
+        serial_interval = self.session_data.get(
+            'interval_storage').iloc[:, 0].values
 
         if data is None:
             raise dash.exceptions.PreventUpdate()
 
-        time_label, inc_label = data.columns[:2]
+        time_label, inc_label = (data.columns[0], 'Incidence Number')
         times = data[time_label]
 
         # Make a new dataframe to save the simulation result
         simulations = data[[time_label]]
 
         # Add the correct R profile to the branchpro model
-        br_pro_model = bp.BranchProModel(new_r0, np.array([1, 2, 3, 2, 1]))
+        if 'Imported Cases' in data.columns:
+            br_pro_model = bp.LocImpBranchProModel(
+                new_r0, serial_interval, new_epsilon)
+            br_pro_model.set_imported_cases(
+                times, data.loc[:, ['Imported Cases']].squeeze().tolist())
+        else:
+            br_pro_model = bp.BranchProModel(new_r0, serial_interval)
         br_pro_model.set_r_profile([new_r0, new_r1], [0, new_t1])
 
         # Generate one simulation trajectory from this model
         simulation_controller = bp.SimulationController(
-            br_pro_model, 1, max(times))
+            br_pro_model, min(times), max(times))
         try:
             data = simulation_controller.run(new_init_cond)
         except ValueError:
             data = -np.ones(max(times))
 
         # Add data to simulations storage
+        print(times, len(times), len(data))
         simulations.loc[:, inc_label] = data
```

Comment:
This line will fail whenever a data file with missing time points is uploaded. I think this is the easiest fix:
```suggestion
        sim_times = simulation_controller.get_regime()
        simulations = pd.DataFrame({time_label: sim_times, inc_label: data})
```